ConsoleApp/Stack.cs

```diff
@@ -0,0 +1,53 @@
+namespace ConsoleApp;
+
+class Stack<T>
+{
+    private record StackItem
+    {
+        public T Value { get; init; }
+        public StackItem Next { get; set; }
+    }
+
+    private StackItem _head;
+    public int Count { get; private set; }
+
+    public void Push(T value)
+    {
+        var newItem = new StackItem
+        {
+            Value = value
+        };
+        newItem.Next = _head;
+        _head = newItem;
+        Count++;
+    }
+
+    public T Pop()
+    {
+        var removedItem = _head;
+        _head = _head.Next;
+        Count--;
+        return removedItem.Value;
+
+    }
+
+    public void Clear()
+    {
+        while (_head != null)
```

Comment:
you can clear stack simply by assigning _head = null. and you forget to assign 0 to Count